torchdrivesim/mesh.py

```diff
@@ -235,28 +236,45 @@ def unpickle(cls, mesh_file_path: str, pickle_module: Any = default_pickle_modul
             return road_mesh
         else:
             raise BadMeshFormat
+    def serialize(self):
+        return {
+            'verts': self.verts.tolist(),
+            'faces': self.faces.tolist()
+        }
 
     def save(self, file_save_path: str):
         """
-        Save the attributes of the object in a non-pickle format.
+        Save the attributes of the mesh object in a json where tensors are converted to lists.
         """
-        if not os.path.exists(os.path.dirname(file_save_path)):
+        directory = os.path.dirname(file_save_path)
+        if directory and not os.path.exists(directory):
             os.makedirs(os.path.dirname(file_save_path))
```

Comment:
This is now `os.makedirs(directory)`. I would also set `exists_ok=True`, just in case the directory is created concurrently.